native/index.d.ts

```diff
@@ -0,0 +1,68 @@
+// <reference path="../react/react-native.d.ts" />
+
+declare module "styled-components/native" {
+    import * as ReactNative from "react-native";
+    import { StatelessComponent } from "react";
+
+    interface StyledFunction<T> {
+      (values: TemplateStringsArray): T;
+      (...values: any[]): T;
```

Comment:
I'm only glancing at the picture you've posted, but I think you want something like

```ts
(strs: TemplateStringsArray, ...fns: Array<(props: any) => string>): T;
```